File: app/src/main/java/com/evaluation/dagger/RestAdapterModule.java
package com.evaluation.dagger;

import com.evaluation.network.RestAdapter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
@Module
public class RestAdapterModule {

    private static RestAdapterModule instance;

    RestAdapterModule() {
    }

    public static RestAdapterModule getInstance() {
        if (instance == null) {
            instance = new RestAdapterModule();
        }
        return instance;
    }

    @Provides
    @Singleton
    RestAdapter provideControlCenterRetrofitAdapter() {
        return new RestAdapter();
    }

}

File: app/src/main/java/com/evaluation/dagger/data/RestAdapterModule.kt
package com.evaluation.dagger.data

import com.evaluation.network.RestAdapter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Vladyslav Havrylenko
 * @since 10.03.2020
 */
@Module
object  RestAdapterModule {
    @Provides
    @Singleton
    fun provideRestAdapter(): RestAdapter = RestAdapter()
}